src/backend/web-gl/function-node.js

```diff
@@ -11,6 +11,10 @@ const constantsPrefix = 'this.constants.';
 const DECODE32_ENCODE32 = /decode32\(\s+encode32\(/g;
 const ENCODE32_DECODE32 = /encode32\(\s+decode32\(/g;
 
+// these debugs were hugely usefull...
+// TODO: optimise out - webpack/babel options? maybe some generic logging support in core/utils?
+// const debug_log = console.log
+const debug_log = () => {};
```

Comment:
I think it really depends on what is being debugged.  We generally remove debugging as much of this can be done using modern debugging tools.
At the very least, it should follow casing already defined `debugLog`.  If we used it as a method, we could check if debug is enabled inside it, and only do something if debug as enabled.  Example:

```js
debugLog(msg) {
  if (!this.debug) return;
  // do stuff here
}
```

Then later:
```js
this.debug('something');
```

However, this really is outside the scope of this issue, and if it were introduced, it'd be best to open an issue, and address it there.  In the mean time, removing it here.

Reply:
it is 'removed' here. its just an empty function...you need to uncomment line 16 for debug to be enabled!

I don't know any modern debugger features that can give this kind of info without lots of faffing about - you just don't get the 'history' needed.  It was hugely useful again for fixing #300, just switched it on and the problem easy to find...

so I'm pretty reticent about removing it as I know I will need to stick it back in again!